src/core/tests/pattern.cpp

```diff
@@ -508,6 +510,68 @@ TEST(pattern, matching_optional) {
                          std::make_shared<op::v0::Abs>(c)));
 }
 
+TEST(pattern, optional_full_match) {
+    Shape shape{};
+    auto model_input1 = std::make_shared<op::v0::Parameter>(element::i32, shape);
+    auto model_input2 = std::make_shared<op::v0::Parameter>(element::i32, shape);
+    auto model_add = std::make_shared<op::v1::Add>(model_input1->output(0), model_input2->output(0));
+    auto model_relu = std::make_shared<op::v0::Relu>(model_add->output(0));
+
+    auto pattern_add = ov::pass::pattern::optional<op::v1::Add>();
+    auto pattern_relu = std::make_shared<op::v0::Relu>(pattern_add->output(0));
+
+    TestMatcher tm;
+
+    ASSERT_TRUE(tm.match(pattern_relu, model_relu));
+}
+
+TEST(pattern, optional_half_match) {
+    Shape shape{};
+    auto model_input1 = std::make_shared<op::v0::Parameter>(element::i32, shape);
+    auto model_input2 = std::make_shared<op::v0::Parameter>(element::i32, shape);
+    auto model_add = std::make_shared<op::v1::Add>(model_input1->output(0), model_input2->output(0));
+    auto model_relu = std::make_shared<op::v0::Relu>(model_add->output(0));
+
+    auto pattern_relu = ov::pass::pattern::optional<op::v0::Relu>();
+    auto pattern_relu1 = std::make_shared<op::v0::Relu>(pattern_relu->output(0));
+
+    TestMatcher tm;
+
+    ASSERT_TRUE(tm.match(pattern_relu1, model_relu));
+}
+
+TEST(pattern, optional_testing) {
```

Comment:
in OV we have some operations with "optional" inputs, e.g. Interpolate-11 operation
https://docs.openvino.ai/2023.3/openvino_docs_ops_image_Interpolate_11.html#

<img width="550" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/04d42d93-1d3d-467a-8b47-430a80e50734">

optional class might be used to match Interpolate with different number of inputs: 2 inputs and 3 inputs using 1 instance:
Interpolate(wrap_type, wrap_type, optional)

could you check if we have such a test? if not could you add it?